decidim-assemblies/app/controllers/decidim/assemblies/assemblies_controller.rb

```diff
@@ -22,7 +22,9 @@ def index
         authorize! :read, Assembly
       end
 
-      def show; end
+      def show
+        redirect_to "/404" unless current_user_can_visit_space?
```

Comment:
Is this returning the 404 Not Found HTTP status?

Reply:
Now we make a redirect to the index view and show a flash alert telling that are not allowed to view the content. it's going well?